### PR TITLE
IBX-6410 PriceQuery documented

### PR DESCRIPTION
**Pull request by dabrt**
| Question      | Answer |
| ------------- | --- |
| JIRA Ticket   | [IBX-6410](https://issues.ibexa.co/browse/IBX-6410) |
| Versions      | master |
| Edition       | all |

Add a description of the PriceQuery with its criteria, add PriceResolver to the API documentation

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date
- [x] Code samples are working
- [x] PHP code samples have been fixed with PHP CS fixer
- [x] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: code_samples/api/product_catalog/src/Command/ProductPriceCommand.php [hunk 1]
```diff
@@ -82,6 +86,17 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             $output->writeln($price);
         }
 
+        $priceCriterions = [
```
Comment (konradoboza) [inline code_samples/api/product_catalog/src/Command/ProductPriceCommand.php:89]:
```suggestion
        $priceCriteria = [
```

File: code_samples/api/product_catalog/src/Command/ProductPriceCommand.php [hunk 2]
```diff
@@ -82,6 +86,17 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+            new Currency('USD'),
+            new CustomerGroup('customer_group_1'),
+            new Product('ergo_desk'),
+        ];
+
+        $priceQuery = new PriceQuery(new LogicalOr(...$priceCriterions));
```
Comment (konradoboza) [inline code_samples/api/product_catalog/src/Command/ProductPriceCommand.php:95]:
```suggestion
        $priceQuery = new PriceQuery(new LogicalOr(...$priceCriteria));
```

File: docs/search/criteria_reference/price_currency_criterion.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+---
+description: Price Currency Criterion
+---
+
+# Price Currency Criterion
+
+The `Currency` Search Criterion searches for prices based on the currency code.
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_currency_criterion.md:7]:
```suggestion
The `Currency` Search Criterion searches for prices based on the given currency.
```

File: docs/search/criteria_reference/price_currency_criterion.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+## Arguments
+
+- `currency_id` - a string currency code
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_currency_criterion.md:11]:
```suggestion
- `currency` - Ibexa\Contracts\ProductCatalog\Values\CurrencyInterface or Ibexa\Contracts\ProductCatalog\Values\CurrencyInterface[]
```

File: docs/search/criteria_reference/price_currency_criterion.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+## Example
+
+### PHP
+
+``` php
+$query = new PriceQuery( 
+    new \Ibexa\Contracts\ProductCatalog\Values\Price\Query\Criterion\Currency('USD')
+);
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_currency_criterion.md:20]:
```suggestion
$currency = $priceService->getPriceById(123);
$query = new PriceQuery( 
    new \Ibexa\Contracts\ProductCatalog\Values\Price\Query\Criterion\Currency($currency)
);
```

File: docs/search/criteria_reference/price_customergroup_criterion.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+description: Price CustomerGroup Criterion
+---
+
+# Price CustomerGroup Criterion
+
+The `CustomerGroup` Search Criterion searches for prices based on the customer group ID.
+
+## Arguments
+
+- `customer_group_id` - an int representing the customer group ID
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_customergroup_criterion.md:11]:
```suggestion
- `customer_group` - Ibexa\Contracts\ProductCatalog\Values\CustomerGroupInterface or Ibexa\Contracts\ProductCatalog\Values\CustomerGroupInterface[]
```

File: docs/search/criteria_reference/price_customergroup_criterion.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+## Example
+
+### PHP
+
+``` php
+$query = new PriceQuery( 
+    new \Ibexa\Contracts\ProductCatalog\Values\Price\Query\Criterion\CustomerGroup(1)
+);
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_customergroup_criterion.md:20]:
```suggestion
$customerGroup = $customerGroupService->getCustomerGroup(123);
$query = new PriceQuery( 
    new \Ibexa\Contracts\ProductCatalog\Values\Price\Query\Criterion\CustomerGroup($customerGroup)
);
```

File: docs/search/criteria_reference/price_isbaseprice_criterion.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+The `IsBasePrice` Criterion is not available in Solr or Elasticsearch engines.
+
+## Example
+
+### PHP
+
+``` php
+$query = new PriceQuery( 
+    new \Ibexa\Contracts\ProductCatalog\Values\Price\Query\Criterion\IsBasePrice('base', true)
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_isbaseprice_criterion.md:24]:
```suggestion
    new \Ibexa\Contracts\ProductCatalog\Values\Price\Query\Criterion\IsBasePrice()
```

File: docs/search/criteria_reference/price_isbaseprice_criterion.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+---
+
+# Price IsBasePrice Criterion
+
+The `IsBasePrice` Search Criterion searches for prices that are base prices.
+
+## Arguments
+
+- `type` - string that represents a price type
+- `base` - Boolean that represents whether to search for base prices or non-base prices
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_isbaseprice_criterion.md:12]: There are no arguments for this criterion.
```suggestion
```

File: docs/search/criteria_reference/price_iscustomprice_criterion.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+---
+
+# Price IsCustomPrice Criterion
+
+The `IsCustomPrice` Search Criterion searches for prices that are custom prices.
+
+## Arguments
+
+- `type` - string that represents a price type
+- `customer_group` - string that represents a Customer Group code that the custom price applies to
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_iscustomprice_criterion.md:12]:
```suggestion
```

File: docs/search/criteria_reference/price_iscustomprice_criterion.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+
+The `IsCustomPrice` Criterion is not available in Solr or Elasticsearch engines.
+
+## Example
+
+### PHP
+
+``` php
+$query = new PriceQuery( 
+    new \Ibexa\Contracts\ProductCatalog\Values\Price\Query\Criterion\IsCustomPrice('custom', 'customer_group_1')
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_iscustomprice_criterion.md:24]:
```suggestion
    new \Ibexa\Contracts\ProductCatalog\Values\Price\Query\Criterion\IsCustomPrice()
```

File: docs/search/criteria_reference/price_logicaland_criterion.md [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+
+## Example
+
+### PHP
+
+``` php
+$query->query = new \Ibexa\Contracts\ProductCatalog\Values\Price\Query\Criterion\LogicalAnd(
+    [
+        new \Ibexa\Contracts\ProductCatalog\Values\Price\Query\Criterion\Currency('USD'),
+        new \Ibexa\Contracts\ProductCatalog\Values\Price\Query\Criterion\IsCustomPrice('custom', 'customer_group_1')
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_logicaland_criterion.md:22]:
```suggestion
        new \Ibexa\Contracts\ProductCatalog\Values\Price\Query\Criterion\IsCustomPrice()
```

File: docs/search/criteria_reference/price_logicalor_criterion.md [hunk 13]
```diff
@@ -0,0 +1,25 @@
+---
+description: Shipment LogicalOr Criterion
+edition: commerce
+---
+
+# Shipment LogicalOr Criterion
+
+The `LogicalOr` Search Criterion matches shipments if at least one of the provided Criteria matches.
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_logicalor_criterion.md:8]:
```suggestion
The `LogicalOr` Search Criterion matches prices if at least one of the provided Criteria matches.
```

File: docs/search/criteria_reference/price_product_criterion.md [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+description: Price Product Criterion
+---
+
+# Price Product Criterion
+
+The `Product` Search Criterion searches for prices based on the product code.
+
+## Arguments
+
+- `product_code` - a string that represents a product code
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_product_criterion.md:11]:
```suggestion
- `product_code` - a string that represents a product code or array of codes
```

File: docs/search/criteria_reference/price_product_criterion.md [hunk 15]
```diff
@@ -0,0 +1,21 @@
+---
+description: Price Product Criterion
+---
+
+# Price Product Criterion
+
+The `Product` Search Criterion searches for prices based on the product code.
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_product_criterion.md:7]:
```suggestion
The `Product` Search Criterion searches for prices based on the product codes.
```

File: docs/search/criteria_reference/price_search_criteria.md [hunk 16]
```diff
@@ -0,0 +1,22 @@
+---
+description: Price Search Criteria
+page_type: reference
+---
+
+# Price Search Criteria reference
+
+Price Search Criteria are part of [Search Criteria](search_criteria_reference.md) and they are supported only by Price Search.
+
+With these Criteria you can filter prices by their payment identifier, payment creation date, payment status, payment method, order, and so on.
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_search_criteria.md:10]: To much of copy&paste.

File: docs/search/criteria_reference/price_search_criteria.md [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+
+|Search Criterion|Search based on|
+|-----|-----|
+|[Currency](price_currency_criterion.md)|Currency code|
+|[CustomerGroup](price_customergroup_criterion.md)|A customer group that the price applies to|
+|[IsBasePrice](price_isbaseprice_criterion.md)|Boolean that indicates whether the price is a base price|
+|[IsCustomPrice](price_iscustomprice_criterion.md)|Boolean that indicates whether the price is a custom price|
+|[LogicalAnd](price_logicaland_criterion.md)|Logical AND criterion that matches if all the provided Criteria match|
+|[LogicalOr](price_logicalor_criterion.md)|Logical OR criterion that matches if at least one of the provided Criteria matches|
+|[Product](price_product_criterion.md)|ID of an associated product|
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_search_criteria.md:22]:
```suggestion
|[Product](price_product_criterion.md)|code of an associated product|
```

File: docs/search/criteria_reference/price_search_criteria.md [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+
+Price Search Criteria are part of [Search Criteria](search_criteria_reference.md) and they are supported only by Price Search.
+
+With these Criteria you can filter prices by their payment identifier, payment creation date, payment status, payment method, order, and so on.
+
+## Payment Search Criteria
+
+|Search Criterion|Search based on|
+|-----|-----|
+|[Currency](price_currency_criterion.md)|Currency code|
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_search_criteria.md:16]:
```suggestion
|[Currency](price_currency_criterion.md)|Currency|
```

**Review by konradoboza — CHANGES_REQUESTED** [review]

File: code_samples/api/product_catalog/src/Command/ProductPriceCommand.php [hunk 19]
```diff
@@ -18,6 +24,8 @@ final class ProductPriceCommand extends Command
 {
     private ProductPriceServiceInterface $productPriceService;
 
+    private PriceResolverInterface $priceResolver;
```
Comment (konradoboza) [inline code_samples/api/product_catalog/src/Command/ProductPriceCommand.php:27]: We are missing constructor parameter and its assignment:
```
public function __construct(
    CurrencyServiceInterface $currencyService,
    ProductServiceInterface $productService,
    ProductPriceServiceInterface $productPriceService,
    PriceResolverInterface $priceResolver,
    UserService $userService,
    PermissionResolver $permissionResolver
) {
    $this->currencyService = $currencyService;
    $this->productPriceService = $productPriceService;
    $this->priceResolver = $priceResolver.
    $this->productService = $productService;
    $this->userService = $userService;
    $this->permissionResolver = $permissionResolver;
 
    parent::__construct('doc:price');
}

File: docs/search/criteria_reference/price_currency_criterion.md [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+description: Price Currency Criterion
+---
+
+# Price Currency Criterion
+
+The `Currency` Search Criterion searches for prices based on the given currency.
+
+## Arguments
+
+- `currency` - a `CurrencyInterface` object that represents the currency (`Ibexa\Contracts\ProductCatalog\Values\CurrencyInterface`)
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_currency_criterion.md:11]: As mentioned, it also accepts array of currencies.

File: docs/search/criteria_reference/price_currency_criterion.md [hunk 21]
```diff
@@ -0,0 +1,23 @@
...
+
+## Example
+
+### PHP
+
+``` php
+$currency = $priceService->getPriceById(123);
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_currency_criterion.md:18]: That's my mistake, it should be:
```suggestion
$currency = $currencyService->getCurrencyByCode('EUR');
```

File: docs/search/criteria_reference/price_customergroup_criterion.md [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+description: Price CustomerGroup Criterion
+---
+
+# Price CustomerGroup Criterion
+
+The `CustomerGroup` Search Criterion searches for prices based on the customer group.
+
+## Arguments
+
+- `customer_groupgroup` - a `CustomerGroupInterface` object that represents the customer group (`Ibexa\Contracts\ProductCatalog\Values\CustomerGroupInterface`)
```
Comment (konradoboza) [inline docs/search/criteria_reference/price_customergroup_criterion.md:11]:
```suggestion
- `customer_group` - a `CustomerGroupInterface` object that represents the customer group (`Ibexa\Contracts\ProductCatalog\Values\CustomerGroupInterface`)
```
besides, it also works with array of customer groups.

File: docs/search/criteria_reference/price_search_criteria.md (hunk 17 above)
Comment (konradoboza) [reply docs/search/criteria_reference/price_search_criteria.md:22]: This hasn't been fixed.

**Review by konradoboza — CHANGES_REQUESTED** [review]

File: code_samples/api/product_catalog/src/Command/ProductPriceCommand.php (hunk 19 above)
Comment (dabrt) [reply code_samples/api/product_catalog/src/Command/ProductPriceCommand.php:27]: Please see the updated section below

**Review by konradoboza — APPROVED** [review]

File: mkdocs.yml [hunk 23]
```diff
@@ -525,6 +525,15 @@ nav:
             - LogicalOr: search/criteria_reference/payment_method_logicalor_criterion.md
             - Type: search/criteria_reference/payment_method_type_criterion.md
             - UpdatedAt: search/criteria_reference/payment_method_updatedat_criterion.md
+        - Price Search Criteria: 
+            - Price Search Criteria: search/criteria_reference/price_search_criteria.md
+            - Currency: search/criteria_reference/price_currency_criterion.md
+            - CustomerGrup: search/criteria_reference/price_customergroup_criterion.md
+            - IsBasePrice: search/criteria_reference/price_isbaseprice_criterion.md
+            - IsCustomPrice: search/criteria_reference/price_iscustomprice_criterion.md
+            - LogicalAnd: search/criteria_reference/payment_logicaland_criterion.md
```
Comment (juskora) [inline mkdocs.yml:534]:
```suggestion
            - LogicalAnd: search/criteria_reference/price_logicaland_criterion.md
```

File: docs/search/criteria_reference/price_logicalor_criterion.md [hunk 24]
```diff
@@ -0,0 +1,25 @@
+---
+description: Shipment LogicalOr Criterion
```
Comment (juskora) [inline docs/search/criteria_reference/price_logicalor_criterion.md:2]:
```suggestion
description: Price LogicalOr Criterion
```

File: docs/search/criteria_reference/price_logicalor_criterion.md [hunk 25]
```diff
@@ -0,0 +1,25 @@
...
+edition: commerce
+---
+
+# Shipment LogicalOr Criterion
```
Comment (juskora) [inline docs/search/criteria_reference/price_logicalor_criterion.md:6]:
```suggestion
# Price LogicalOr Criterion
```

File: mkdocs.yml [hunk 26]
```diff
@@ -525,6 +525,15 @@ nav:
...
+            - LogicalOr: search/criteria_reference/payment_logicalor_criterion.md
```
Comment (juskora) [inline mkdocs.yml:535]:
```suggestion
            - LogicalOr: search/criteria_reference/price_logicalor_criterion.md
```

File: docs/search/criteria_reference/price_search_criteria.md [hunk 27]
```diff
@@ -0,0 +1,22 @@
...
+page_type: reference
+---
+
+# Price Search Criteria reference
+
+Price Search Criteria are part of [Search Criteria](search_criteria_reference.md) and they are supported only by Price Search.
+
+With these Criteria you can filter prices by currency, customer group, product, and so on.
+
+## Payment Search Criteria
```
Comment (juskora) [inline docs/search/criteria_reference/price_search_criteria.md:12]: Payment or Price?

**Review by juskora — CHANGES_REQUESTED** [review]

File: docs/pim/price_api.md [hunk 28]
```diff
@@ -57,6 +66,25 @@ For example, to create a new price for a given currency, use `ProductPriceServic
     That is why all amounts are provided [in the smallest unit](https://www.moneyphp.org/en/stable/getting-started.html#instantiation).
     For example, for euro `50000` refers to 50000 cents, equal to 50 euros.
 
+### Resolve prices
+
+To display a product price on a product page or in the cart, you must calculate its value based on a base price and the context, where context is information whether any price modifiers that might be applied in relation to a customer group exist.
+To obtain the final price, or resolve the price, use the `PriceResolverInterface` service, which uses the following logic:
+
+1. Checks whether a price exists for the product and currency, returns null if no such price exists.
```
Comment (juskora) [inline docs/pim/price_api.md:74]: Shouldn't be null put in `` as a data type?

File: docs/pim/price_api.md [hunk 29]
```diff
@@ -57,6 +66,25 @@ For example, to create a new price for a given currency, use `ProductPriceServic
     That is why all amounts are provided [in the smallest unit](https://www.moneyphp.org/en/stable/getting-started.html#instantiation).
     For example, for euro `50000` refers to 50000 cents, equal to 50 euros.
 
+### Resolve prices
+
+To display a product price on a product page or in the cart, you must calculate its value based on a base price and the context, where context is information whether any price modifiers that might be applied in relation to a customer group exist.
```
Comment (MagdalenaZuba) [inline docs/pim/price_api.md:71]:
```suggestion
To display a product price on a product page or in the cart, you must calculate its value based on a base price and the context.
Context contains information about any price modifiers that may apply to a specific customer group.
```

File: docs/pim/price_api.md [hunk 30]
```diff
@@ -57,6 +66,25 @@ For example, to create a new price for a given currency, use `ProductPriceServic
...
+To obtain the final price, or resolve the price, use the `PriceResolverInterface` service, which uses the following logic:
```
Comment (MagdalenaZuba) [inline docs/pim/price_api.md:72]:
```suggestion
To determine the final price, or resolve the price, use the `PriceResolverInterface` service, which uses the following logic:
```

File: docs/search/criteria_reference/price_search_criteria.md (hunk 27 above)
Comment (dabrt) [inline docs/search/criteria_reference/price_search_criteria.md:12]:
```suggestion
## Price Search Criteria
```

**Review by MagdalenaZuba — CHANGES_REQUESTED** [review]

File: docs/pim/price_api.md (hunk 28 above)
Comment (dabrt) [inline docs/pim/price_api.md:74]:
```suggestion
1. Checks whether a price exists for the product and currency, returns `null` if no such price exists.
```

**Review by MagdalenaZuba — APPROVED** [review]

**Review by juskora — APPROVED** [review]

**Review by adriendupuis — APPROVED** [review]
I couldn't run the code itself so I just checked it's logic and format.

**Pull request merged**